### PR TITLE
set isSystem for SVM IPs

### PR DESCRIPTION
**Pull request by DaanHoogland**
### Description

This PR 

Fixes: #9275

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)
- [ ] build/CI
- [ ] test (unit or integration test code)

#### Feature/Enhancement Scale

- [x] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [x] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### CODE REVIEW

**Review by shwstppr — APPROVED** [review]
Code LGTM

File: services/secondary-storage/controller/src/main/java/org/apache/cloudstack/secondarystorage/SecondaryStorageManagerImpl.java [hunk 1]
```diff
@@ -31,6 +31,7 @@
 import javax.inject.Inject;
 import javax.naming.ConfigurationException;
 
+import com.cloud.utils.db.*;
```
Comment (shwstppr) [inline services/secondary-storage/controller/src/main/java/org/apache/cloudstack/secondarystorage/SecondaryStorageManagerImpl.java:34]: Causing build failure

**Review by sureshanaparti — APPROVED** [review]
clgtm

File: services/secondary-storage/controller/src/main/java/org/apache/cloudstack/secondarystorage/SecondaryStorageManagerImpl.java [hunk 2]
```diff
@@ -1309,12 +1308,7 @@ public boolean finalizeStart(VirtualMachineProfile profile, long hostId, Command
 
         try {
             _rulesMgr.getSystemIpAndEnableStaticNatForVm(profile.getVirtualMachine(), false);
-            IPAddressVO ipaddr = _ipAddressDao.findByAssociatedVmId(profile.getVirtualMachine().getId());
-            if (ipaddr != null && ipaddr.getSystem()) {
-                SecondaryStorageVmVO secVm = _secStorageVmDao.findById(profile.getId());
-                secVm.setPublicIpAddress(ipaddr.getAddress().addr());
-                _secStorageVmDao.update(secVm.getId(), secVm);
-            }
```
Comment (sureshanaparti) [inline services/secondary-storage/controller/src/main/java/org/apache/cloudstack/secondarystorage/SecondaryStorageManagerImpl.java:1317]: @DaanHoogland I see the same code in _ConsoleProxyManagerImpl_, should we change there as well?

Comment (DaanHoogland) [reply services/secondary-storage/controller/src/main/java/org/apache/cloudstack/secondarystorage/SecondaryStorageManagerImpl.java:1317]: yes, I wanted to test this part first and then look for a more generic place to put this code

**Comment by weizhouapache** [comment]
thanks @DaanHoogland for your investigation
it is great you found the root cause !

I did a quick search on the code, it looks like the issue is caused by

https://github.com/apache/cloudstack/blob/8806e44e4fb9cc41460ed5cf0ab319be2b018b54/server/src/main/java/com/cloud/network/guru/PublicNetworkGuru.java#L132

(the parameter `false` may should be `forSystemVms`)

**Comment by DaanHoogland** [comment]
> `forSystemVms`

thanks @weizhouapache that does sound like a much better and more generic place to make sure the flag is set. I'll revert my changes here and apply your suggestion.

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by weizhouapache** [comment]
@blueorangutan package

**Review by sureshanaparti — APPROVED** [review]
clgtm

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test alma9 kvm-alma9

**Review by weizhouapache — APPROVED** [review]
code lgtm

verified ok
- apply the change
- destroy a systemvm
- wait until the new system vm is Running
- release the public IP, got an exception as expected
image

**Pull request merged**